src/DDTrace/Integrations/ElasticSearch/V1/ElasticSearchSandboxedIntegration.php

```diff
@@ -0,0 +1,227 @@
+        $this->traceNamespaceMethod('ClusterNamespace', 'getSettings');
+        $this->traceNamespaceMethod('ClusterNamespace', 'health');
+        $this->traceNamespaceMethod('ClusterNamespace', 'pendingTasks');
+        $this->traceNamespaceMethod('ClusterNamespace', 'putSettings');
+        $this->traceNamespaceMethod('ClusterNamespace', 'reroute');
+        $this->traceNamespaceMethod('ClusterNamespace', 'state');
+        $this->traceNamespaceMethod('ClusterNamespace', 'stats');
+
+        // NodesNamespace operations
+        $this->traceNamespaceMethod('NodesNamespace', 'hotThreads');
+        $this->traceNamespaceMethod('NodesNamespace', 'info');
+        $this->traceNamespaceMethod('NodesNamespace', 'shutdown');
+        $this->traceNamespaceMethod('NodesNamespace', 'stats');
+
+        // Endpoints
+        dd_trace_method('Elasticsearch\Endpoints\AbstractEndpoint', 'performRequest', function (SpanData $span, $args) {
+            if (dd_trace_tracer_is_limited()) {
+                return false;
+            }
+            $span->name = "Elasticsearch.Endpoint.performRequest";
+            $span->resource = 'performRequest';
+            $span->service = ElasticSearchSandboxedIntegration::NAME;
+            $span->type = Type::ELASTICSEARCH;
+
+            try {
+                $span->meta[Tag::ELASTICSEARCH_URL] = $this->getURI();
+                $span->meta[Tag::ELASTICSEARCH_METHOD] = $this->getMethod();
+                if (is_array($this->params)) {
+                    $span->meta[Tag::ELASTICSEARCH_PARAMS] = json_encode($this->params);
+                }
+                if ($this->getMethod() === 'GET' && $body = $this->getBody()) {
+                    $span->meta[Tag::ELASTICSEARCH_BODY] = json_encode($this->getBody());
+                }
+            } catch (\Exception $ex) {
+            }
+        });
+
+        return Integration::LOADED;
+    }
+    /**
+     * @param string $name
+     * @param bool $isTraceAnalyticsCandidate
+     */
+    public function traceClientMethod($name, $isTraceAnalyticsCandidate = false)
+    {
+        $integration = $this;
+        $class = 'Elasticsearch\Client';
+
+        dd_trace_method(
+            $class,
+            $name,
+            function (SpanData $span, $args) use ($name, $isTraceAnalyticsCandidate, $integration) {
+                if (dd_trace_tracer_is_limited()) {
+                    return false;
+                }
+                $params = [];
+                if (isset($args[0])) {
+                    list($params) = $args;
+                }
+
+                $span->name = "Elasticsearch.Client.$name";
+
+                if ($isTraceAnalyticsCandidate) {
+                    $integration->addTraceAnalyticsIfEnabled($span);
+                }
+
+                $span->service = ElasticSearchSandboxedIntegration::NAME;
+                $span->type = Type::ELASTICSEARCH;
+                $span->resource = ElasticSearchCommon::buildResourceName($name, $params);
```

Comment:
Bikeshed warning!  ⚠️ 😆 I think this might be a bit more readable than that `$params` temp var above:

```suggestion
                $span->resource = ElasticSearchCommon::buildResourceName($name, isset($args[0]) ? $args[0] : []);
```

I can see this is in the original integration so no worries if you want to keep it as-is. But if you wanna change it, it also exists in `traceNamespaceMethod()` below. :)

Reply:
:palm_face: 😄 nice suggestion